components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/util/FrameworkUtils.java

```diff
@@ -2414,5 +2421,81 @@ public static FederatedAssociationManager getFederatedAssociationManager() throw
         }
         return federatedAssociationManager;
     }
-}
 
+    /**
+     * Retrieves the unique user id of the given username. If the unique user id is not available, generate an id and
+     * update the userid claim in read/write userstores.
+     *
+     * @param tenantDomain    tenant domain of the user
+     * @param userStoreDomain userstore of the user
+     * @param username        username
+     * @return unique user id of the user
+     * @throws UserSessionException
+     */
+    public static String resolveUserIdFromUsername(String tenantDomain, String userStoreDomain, String username) throws
```

Comment:
shall we make this tenant id, since we already have tenant id from the place we invoke this method so that we can avoid the tenant id retrieval again?

Reply:
Fixed